src/audio/module_adapter/module_adapter.c

```diff
@@ -792,19 +778,15 @@ static int module_adapter_simple_copy(struct comp_dev *dev)
 	list_for_item(blist, &dev->bsource_list)
 		buffer_release(source_c[i++]);
 
-	for (i = 0; i < mod->num_output_buffers; i++)
-		mod->output_buffers[i].size = 0;
-
-	for (i = 0; i < mod->num_input_buffers; i++) {
-		mod->input_buffers[i].size = 0;
-		mod->input_buffers[i].consumed = 0;
-	}
-
 	return ret;
 }
 
 int module_adapter_copy(struct comp_dev *dev)
 {
+	struct input_stream_buffer input_buffers[PLATFORM_MAX_STREAMS];
+	struct output_stream_buffer output_buffers[PLATFORM_MAX_STREAMS];
```

Comment:
This was noted in the original review, but this could be a stack size issue given stack sizes are separately configurable. At least we have good stack-overrun tools now.

Reply:
@kv2019i at least I removed identical arrays from the original version https://github.com/thesofproject/sof/pull/7449/commits/32911d654d8e85719f5121a0b584ab77c3210a1f#diff-8065c18eec2f5b971af80cc5aa42ec413fd9432024dfb0040c676c9dee278f31L700-R683 in `module_adapter_simple_copy()` where that function is called from this one so both are allocated on stack at the same time... But yes, even a single set makes up for 384 bytes...